buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/RestResourcesPlugin.java

```diff
@@ -86,21 +86,29 @@
     public void apply(Project project) {
         RestResourcesExtension extension = project.getExtensions().create(EXTENSION_NAME, RestResourcesExtension.class);
 
+        // tests
         Provider<CopyRestTestsTask> copyRestYamlTestTask = project.getTasks()
             .register("copyYamlTestsTask", CopyRestTestsTask.class, task -> {
                 task.includeCore.set(extension.restTests.getIncludeCore());
                 task.includeXpack.set(extension.restTests.getIncludeXpack());
-                task.coreConfig = project.getConfigurations().create("restTest");
+                task.coreConfig = project.getConfigurations().create("restTestConfig");
```

Comment:
We should probalby not be creating these configurations inside this configuration block. The reason being, because we are using the lazy task creation api, that means that _sometimes_ we create these, and sometimes we don't, depending on the requested task graph. When it comes to configuration API elements, we should always create these or we could get funny errors depending on what tasks you ask to run.

Reply:
thanks, implemented on https://github.com/elastic/elasticsearch/pull/53620/commits/287fb72ec4ce5984e41cd69de907b1092863b200